src/index.js

```diff
@@ -54,6 +55,22 @@ export default class Zeroconf extends EventEmitter {
       this.emit('update')
     })
 
+    this._dListeners.published = DeviceEventEmitter.addListener('RNZeroconfServiceRegistered', service => {
+      if (!service || !service.name) { return }
+
+      this._publishedServices[service.name] = service
+      this.emit('registered', service)
+      this.emit('update')
```

Comment:
Because this event is to keep track of current discovered services, publishing doesn't update those, there is no need to call it here